packages/enzyme/src/ShallowWrapper.js

```diff
@@ -360,9 +360,19 @@ class ShallowWrapper {
    * @return {Array<ReactComponent>}
    */
   getNodesInternal() {
+    const adapter = getAdapter(this[OPTIONS]);
+    const prevProps = (this[UNRENDERED] && this[UNRENDERED].props) || {};
+
     if (this[ROOT] === this && this.length === 1) {
-      this.update();
+      if (adapter.shouldUpdateComponent) {
+        if (adapter.shouldUpdateComponent(prevProps, this[ROOT])) {
```

Comment:
this can probably be collapsed into:
```js
if (!adapter.shouldUpdateComponent || adapter.shouldUpdateComponent(prevProps, this[ROOT])) {
  this.update();
}
```